fuel/src/test/kotlin/com/github/kittinunf/fuel/RequestUploadTest.kt

```diff
@@ -235,4 +231,45 @@ class RequestUploadTest : BaseTestCase() {
         val statusCode = HttpURLConnection.HTTP_OK
         assertThat(response?.httpStatusCode, isEqualTo(statusCode))
     }
+
+    @Test
+    fun httpUploadWithDataParts() {
+        var request: Request? = null
+        var response: Response? = null
+        var data: Any? = null
+        var error: FuelError? = null
+
+        manager.upload("/post", param = listOf("foo" to "bar"))
+                .sources { request, url ->
+                    listOf(File(currentDir, "lorem_ipsum_short.tmp"),
+                            File(currentDir, "lorem_ipsum_long.tmp"))
+                }
+                // Override with Data parts
+                .dataParts { request, url ->
+                    listOf(DataPart("first-file", "image/jpeg", File(currentDir, "lorem_ipsum_short.tmp")),
```

Comment:
Oh, actually only `dataParts` is fine. Example show what if we call `dataParts` it will override all files in `sources`.